Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,320 @@
+        /// <returns>A task that contains a <see cref="SyntaxValidationResult"/> object, which contains the structured validation objects and a list of <see cref="ValidationStructuredEntry"/> objects accumulated during the validation.</returns>
+        public static async Task<SyntaxValidationResult> ValidatePxFileMetadataSyntaxAsync(
+            Stream stream,
+            Encoding encoding,
+            string filename,
+            PxFileSyntaxConf? syntaxConf = null,
+            int bufferSize = DEFAULT_BUFFER_SIZE,
+            CustomValidationFunctions? customValidationFunctions = null,
+            CancellationToken cancellationToken = default)
+        {
+            SyntaxValidationFunctions validationFunctions = new();
+            IEnumerable<EntryValidationFunctionDelegate> stringValidationFunctions = validationFunctions.DefaultStringValidationFunctions;
+            IEnumerable<KeyValuePairValidationFunctionDelegate> keyValueValidationFunctions = validationFunctions.DefaultKeyValueValidationFunctions;
+            IEnumerable<StructuredValidationFunctionDelegate> structuredValidationFunctions = validationFunctions.DefaultStructuredValidationFunctions;
+
+            if (customValidationFunctions is not null)
+            {
+                stringValidationFunctions = stringValidationFunctions.Concat(customValidationFunctions.CustomStringValidationFunctions);
+                keyValueValidationFunctions = keyValueValidationFunctions.Concat(customValidationFunctions.CustomKeyValueValidationFunctions);
+                structuredValidationFunctions = structuredValidationFunctions.Concat(customValidationFunctions.CustomStructuredValidationFunctions);
+            }
+
+            syntaxConf ??= PxFileSyntaxConf.Default;
+
+            List<ValidationFeedbackItem> validationFeedback = [];
+            List<ValidationEntry> entries = await BuildValidationEntriesAsync(stream, encoding, syntaxConf, filename, bufferSize, cancellationToken);
+            ValidateEntries(entries, stringValidationFunctions, validationFeedback, syntaxConf);
+            List<ValidationKeyValuePair> keyValuePairs = BuildKeyValuePairs(entries, syntaxConf);
+            ValidateKeyValuePairs(keyValuePairs, keyValueValidationFunctions, validationFeedback, syntaxConf);
+            List<ValidationStructuredEntry> structuredEntries = BuildValidationStructureEntries(keyValuePairs, syntaxConf);
+            ValidateStructs(structuredEntries, structuredValidationFunctions, validationFeedback, syntaxConf);
+
+            return new([.. validationFeedback], structuredEntries);
+        }
+
+        private static void ValidateEntries(IEnumerable<ValidationEntry> objects, IEnumerable<EntryValidationFunctionDelegate> validationFunctions, List<ValidationFeedbackItem> validationFeedback, PxFileSyntaxConf syntaxConf)
+        {
+            foreach (var @object in objects)
```

Comment:
Use explicit types instead of vars. Rename objects to validationEntries or something.